src/System.ValueTuple/src/System/ValueTuple/TupleExtensions.cs

```diff
@@ -920,7 +920,7 @@ public static class TupleExtensions
         }
         #endregion
 
-        private static ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest> CreateLong<T1, T2, T3, T4, T5, T6, T7, TRest>(T1 item1, T2 item2, T3 item3, T4 item4, T5 item5, T6 item6, T7 item7, TRest rest) where TRest : struct =>
+        private static ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest> CreateLong<T1, T2, T3, T4, T5, T6, T7, TRest>(T1 item1, T2 item2, T3 item3, T4 item4, T5 item5, T6 item6, T7 item7, TRest rest) where TRest : struct, ITuple =>
             new ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest>(item1, item2, item3, item4, item5, item6, item7, rest);
```

Comment:
For me, the reason is that we want to constrain to a `ValueTuple`, but we can't express that, so we apply the maximum constrain that we can express.

I find the idea of switching to nesting class (most likely `System.Tuple`) after a certain arity intriguing. But that means the compiler would have to be aware of that type too. Even if we removed the struct constraint to leave room for such an option in the future, we'd face the problem that `ValueTuple` and `System.Tuple` don't implement the same internal interface.